packages/js/components/src/experimental-select-control/select-control.tsx

```diff
@@ -150,6 +158,32 @@ function SelectControl< ItemType = DefaultItemType >( {
 					break;
 			}
 		},
+		stateReducer: ( state, actionAndChanges ) => {
```

Comment:
Instead of creating a new state reducer and `keepMenuOpenOnSelect` prop, could we introduce an `onBlur` and `onFocus` prop to the component?

This gives the consumer a lot more control over the menu interaction and is less opinionated about the state of openness.

We could add a new case to the above `onStateChange` reducer:

```
				case useCombobox.stateChangeTypes.InputBlur:
					onBlur();
```

And for the focus, we can add it to the combobox below:

```
						onFocus: () => {
							setIsFocused( true );
							onFocus();
						},
```

Then in the consumer (the story):

```
const [ isOpen, setIsOpen ] = useState( false );

<SelectControl
	onBlur={ () => setIsOpen( false ) }
	onFocus={ () => setIsOpen( true ) }
>
	<Menu isOpen={ isOpen } getMenuProps={ getMenuProps }>
</SelectControl>
```

Reply:
Yeah I could give this solution a try, one thing I do wonder though if it would cause a conflict between the `isOpen` state of combobox and the `isOpen` state of the outside component.
Currently the only item I can see relying on this is the `getDropdownProps` not sure how important that is. Unless we also allow the option for the user to pass in the `isOpen` state to the `SelectControl` component.
Let me see if I can wrangle something together.